src/components/ec-multiple-values-selection/ec-multiple-values-selection.vue

```diff
@@ -0,0 +1,275 @@
+                  :type="item.icon.type"
+                  :size="24"
+                />
+                <span class="ec-multiple-values-selection__label-text">{{ item.text }}</span>
+              </div>
+            </template>
+          </ec-checkbox>
+        </li>
+
+        <li
+          v-for="item in notSelectedItems"
+          :key="item.value"
+          class="ec-multiple-values-selection__value-wrapper"
+          data-test="ec-multiple-values-selection__value-wrapper"
+        >
+          <ec-checkbox
+            class="ec-multiple-values-selection__checkbox"
+            data-test="ec-multiple-values-selection__checkbox-select"
+            @checked-value-change="onSelect(item)"
+          >
+            <template #label>
+              <div
+                class="ec-multiple-values-selection__label-wrapper"
+              >
+                <ec-icon
+                  v-if="item.icon"
+                  class="ec-multiple-values-selection__icon"
+                  :name="item.icon.name"
+                  :type="item.icon.type"
+                  :size="24"
+                />
+                <span class="ec-multiple-values-selection__label-text">{{ item.text }}</span>
+              </div>
+            </template>
+          </ec-checkbox>
+        </li>
+      </div>
+    </ec-loading>
+  </div>
+</template>
+<script>
+// TODO ONL-4912 !!!don't forget to debounce search!!!
+import EcCheckbox from '../ec-checkbox';
+import EcIcon from '../ec-icon';
+import EcInputField from '../ec-input-field';
+import EcLoading from '../ec-loading';
+
+export default {
+  name: 'EcMultipleValuesSelection',
+  components: {
+    EcCheckbox, EcIcon, EcInputField, EcLoading,
+  },
+  model: {
+    prop: 'value',
+    event: 'change',
+  },
+  props: {
+    items: {
+      type: Array,
+      required: true,
+      default: () => ([]),
+    },
+    value: {
+      type: Array,
+      required: false,
+      default: () => ([]),
+    },
+    loading: {
+      type: Boolean,
+    },
+    error: {
+      type: Error,
+      required: false,
+      default: null,
+    },
+    emptyMessage: {
+      type: String,
+    },
+    emptyIcon: {
+      type: String,
+    },
+    isSearchable: {
+      type: Boolean,
+    }, // TODO with https://fxsolutions.atlassian.net/browse/ONL-4912
+    isSelectAll: {
+      type: Boolean,
+    }, // TODO ONL-4911
+    selectAllText: {
+      type: String,
+    },
+    searchPlaceholder: {
+      type: String,
+    },
+  },
+  data() {
+    return {
+      query: '',
+    };
+  },
+  computed: {
+    selectedItems() {
+      return this.value;
+    },
+    notSelectedItems() {
+      const itemsSet = new Set(this.selectedItems.map(item => item.value));
+      return this.items.filter(item => !itemsSet.has(item.value));
+    },
+    // TODO ONL-4911
+    allItemsAreSelected() {
+      return this.notSelectedItems.length === 0;
+    },
+    // TODO ONL-4919
+    noResults() {
```

Comment:
I'm pretty sure we can split this with the `emptyMessage` and `isEmpty` ? ps: naming to be decided